kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java

```diff
@@ -100,17 +100,29 @@ public interface MessageHandler {
   private final class ListenerStream {
     private MessageHandler handler;
     private ExecWatchInputStream inputStream;
+    private String name;
+
+    public ListenerStream(String name) {
+      this.name = name;
+    }
 
     private void handle(ByteBuffer byteString, WebSocket webSocket) throws IOException {
       if (handler != null) {
         handler.handle(byteString);
       } else {
+        if (LOGGER.isDebugEnabled()) {
+          String message = ExecWebSocketListener.toString(byteString);
+          if (message.length() > 200) {
+            message = message.substring(0, 197) + "...";
+          }
+          LOGGER.debug("exec message received on channel {}: {}", name, message);
+        }
         webSocket.request();
       }
     }
   }
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(ExecWebSocketListener.class);
+  static final Logger LOGGER = LoggerFactory.getLogger(ExecWebSocketListener.class);
```

Comment:
Could we keep this private and retrieve the logger in PodOperationContext using the same approach?

Reply:
Also, are we sure that this is the class/package/point where we want to add the flag to decide if these messages should be logged? Can we document how to enable this logging? Would it make sense to check for debug enabled at the base package?

Reply:
> Could we keep this private and retrieve the logger in PodOperationContext using the same approach?

You mean enable this off of two separate loggers?  I didn't like that approach, even if they share a common base.

> Also, are we sure that this is the class/package/point where we want to add the flag to decide if these messages should be logged? 

Yes, they should be logged in the ExecWebSocketListener, and we have to make an earlier decision to include the query flags.

> Can we document how to enable this logging?

Yes, we just have to agree on what it should be first.

> Would it make sense to check for debug enabled at the base package?

You could, it's just a bit more code to create / access a couple of additional loggers - call me lazy, but I'd rather reuse something that exists if possible.

Reply:
> You mean enable this off of two separate loggers? I didn't like that approach, even if they share a common base.

I meant doing this from `PodOperationContext`:
```
boolean debug = // ExecWebSocketListener.LOGGER.isDebugEnabled();
  LoggerFactory.getLogger(ExecWebSocketListener.class);
``` 
It's still coupled, but at least we don't access the other class variable that should be used for internal purposes only.

However, I'm going over both https://github.com/fabric8io/kubernetes-client/pull/4288#discussion_r928781328 and https://github.com/fabric8io/kubernetes-client/pull/4288#discussion_r928777707 and as I see it's just that the structure is flawed.

The `PodOperationContext` is not only used for Exec operations, but for other operations too.

> It's usage should be clear from the context :)

For me it's not clear that a **public** `addQueryParameters` method is used only to add exec-related parameters. Maybe this method just doesn't belong here.

I guess that at this point it doesn't matter anyway.